components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/handler/transaction/TransactionCountHandlerComponent.java

```diff
@@ -44,15 +45,18 @@
 public class TransactionCountHandlerComponent {
 
     private static final Log LOG = LogFactory.getLog(TransactionCountHandlerComponent.class);
-    private static final int DEFAULT_SCHEDULED_PERIOD_INTERVAL = 60;
+    private static final int DEFAULT_UPDATE_INTERVAL = 1;
```

Comment:
Are we going to update every minute in a production server by default , isn't that too much . what about 15 minutes or something. Did we make this decision consciously ?

Reply:
one minute was chosen to reduce the transaction count loss in a situation where a server shut down.

Reply:
One minute was chosen due to the following reasons. 
1. Assuming that we have a tps of 100, we will miss 100*60*5=30K requests if we configure the interval to 5 mins.
2. This is not affecting the direct messaging flow.
3. The performance impact of a simple db update per 1 min is negelgible,